packages/idempotency/README.md

```diff
@@ -1,71 +1,131 @@
+### Decorator function
+
+```ts
+import { makeFunctionIdempotent, DynamoDBPersistenceLayer } from "@aws-lambda-powertools/idempotency";
+
+const dynamoDBPersistenceLayer = new DynamoDBPersistenceLayer();
+
+class MyLambdaHandler implements LambdaInterface {
+  
+  public async handler(_event: Event, _context: Context): Promise<void> {
+    for(const record of _event.Records) {
+      await this.processRecord(record);
+    }
+  }
+  
+  @idempotentFunction({ persistenceStore: dynamoDBPersistenceLayer, dataKeywordArgument: "payload" })
+  public async process(payload: Record<string, unknown>): Promise<void> {
+    // your lambda code here
+  }
+}
+```
 
-👉 [Installation guide for the **Tracer** utility](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/tracer#getting-started)
+The `dataKeywordArgument` parameter is optional. If not provided, the whole event will be used as the idempotency key.
+Otherwise, you need to specify the string name of the argument that contains the data to be used as the idempotency key.
+For example if you have an input like this:
 
-👉 [Installation guide for the **Logger** utility](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger#getting-started)
 
-👉 [Installation guide for the **Metrics** utility](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/metrics#getting-started)
+```json
+{
+  "transactionId": 1235,
+  "product": "book",
+  "quantity": 1,
+  "price": 10
+}
+```
+
+You can use `transactionId` as the idempotency key. This will ensure that the same transaction is not processed twice.
 
-👉 [Installation guide for the **Parameters** utility](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/utilities/parameters/#getting-started)
+### Function wrapper
 
-### Examples
+In case where you don't use classes and decorators you can wrap your function to make it idempotent.
 
-* [CDK](https://github.com/awslabs/aws-lambda-powertools-typescript/tree/main/examples/cdk)
-* [SAM](https://github.com/awslabs/aws-lambda-powertools-typescript/tree/main/examples/sam)
+```ts
+import { makeFunctionIdempotent, DynamoDBPersistenceLayer } from "@aws-lambda-powertools/idempotency";
 
-### Serverless TypeScript Demo application
+const dynamoDBPersistenceLayer = new DynamoDBPersistenceLayer();
+const processingFunction = async (payload: Record<string, unknown>): Promise<void> => {
+  // your lambda code here
+};
 
-The [Serverless TypeScript Demo](https://github.com/aws-samples/serverless-typescript-demo) shows how to use Powertools for AWS Lambda (TypeScript).  
-You can find instructions on how to deploy and load test this application in the [repository](https://github.com/aws-samples/serverless-typescript-demo).
+const processIdempotently = makeFunctionIdempotent(proccessingFunction, {
+  persistenceStore: dynamoDBPersistenceLayer,
+  dataKeywordArgument: "transactionId"
+});
+
+export const handler = async (
+  _event: Event,

```

Comment:
Ideally these examples should be copy-paste-able. If we I do that I don't have imports for `Event` & `Context`.

We can either add the imports from `aws-lambda`, or use `unknown` like done in the rest of the docs/examples.

Reply:
you are right, I always complained about missing import in any other docs, now I did the same 😅